docs/docs/dev_docs/testing/cheat_codes.md

```diff
@@ -0,0 +1,458 @@
+
+### timestamp
+
+#### Function Signature
+```js
+public async timestamp(): Promise<number> 
+```
+
+#### Description 
+Fetches the current Ethereum timestamp.
+
+#### Example
+```js 
+const timestamp = await cc.eth.timestamp()
+```
+
+### mine
+
+#### Function Signature
+```js
+public async mine(numberOfBlocks = 1): Promise<void> 
+```
+
+#### Description
+Mines the specified number of blocks on Ethereum (default 1).
+
+#### Example
+```js
+const blockNum = await cc.eth.blockNumber();
+await cc.eth.mine(10) // mines 10 blocks
+const newBlockNum = await cc.eth.blockNumber(); // = blockNum + 10.
+```
+
+### setNextBlockTimestamp
+
+#### Function Signature  
+```js
+public async setNextBlockTimestamp(timestamp: number): Promise<void>
+```
+
+#### Description
+Sets the timestamp (unix format in seconds) for the next mined block on Ethereum.
+Remember that timestamp can only be set in the future and not in the past.
+
+#### Example
+```js
+// // Set next block timestamp to 16 Aug 2023 10:54:30 GMT
+await cc.eth.setNextBlockTimestamp(1692183270) 
+// next transaction you will do will have the timestamp as 1692183270
+```
+
+### dumpChainState
+
+#### Function Signature
+```js
+public async dumpChainState(fileName: string): Promise<void> 
+```
+
+#### Description
+Dumps the current Ethereum chain state to a file.
+Stores a hex string representing the complete state of the chain in a file with the provided path. Can be re-imported into a fresh/restarted instance of Anvil to reattain the same state.
+When combined with `loadChainState()` cheatcode, it can be let you easily import the current state of mainnet into the Anvil instance of the sandbox, to use Uniswap for example.
+
+#### Example
+```js
+await cc.eth.dumpChainState('chain-state.json') 
+```
+
+### loadChainState
+
+#### Function Signature 
+```js
+public async loadChainState(fileName: string): Promise<void>
+```
+
+#### Description
+Loads the Ethereum chain state from a file which contains a hex string representing an Ethereum state. 
+When given a file previously written to by `cc.eth.dumpChainState()`, it merges the contents into the current chain state. Will overwrite any colliding accounts/storage slots.
+
+#### Example  
+```js
+await cc.eth.loadChainState('chain-state.json')
+```
+
+### load  
+
+#### Function Signature
+```js
+public async load(contract: EthAddress, slot: bigint): Promise<bigint>
+```
+
+#### Description  
+Loads the value at a storage slot of a Ethereum contract.
+
+#### Example
+```
+/// contract LeetContract {
+///     uint256 private leet = 1337; // slot 0
+/// }
+
+const leetContractAddress = EthAddress.fromString('0x1234...');
+const value = await cc.eth.load(leetContractAddress, BigInt(0));
+console.log(value); // 1337
+```
+
+### store
+
+#### Function Signature
+```js
+public async store(contract: EthAddress, slot: bigint, value: bigint): Promise<void>  
+```
+
+#### Description
+Stores the value in storage slot on a Ethereum contract.
+
+#### Example
+```
```

Comment:
probably worth annotating the solidity contracts with `sol and using separate code blocks